library/core/src/iter/traits/iterator.rs

```diff
@@ -81,8 +81,8 @@ fn _assert_is_object_safe(_: &dyn Iterator<Item = ()>) {}
     ),
     on(
         _Self = "[]",
-        label = "borrow the array with `&` or call `.iter()` on it to iterate over it",
-        note = "arrays are not iterators, but slices like the following are: `&[1, 2, 3]`"
```

Comment:
Any reason _not_ to suggest the `&`?

Reply:
hmm, i wasn't sure how to explain that in a concise way, as adding `&` changes the element type.

Do you have an idea for a second label here?